packages/styled-base/src/index.js

```diff
@@ -211,12 +211,11 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
       nextTag: ElementType,
       nextOptions?: StyledOptions
     ) => {
-      return createStyled(
-        nextTag,
-        nextOptions !== undefined
-          ? { ...(options || {}), ...nextOptions }
-          : options
-      )(...styles)
+      return createStyled(nextTag, {
+        shouldForwardProp,
+        ...options,
+        ...nextOptions
+      })(...styles)
```

Comment:
This was done just to simplify the object spreads.

Reply:
There are 2 potential problems I can see here:
1. `shouldForwardProp` contains composed function, but you can override it in here with the one in `options` (so with the one pre-composition)
2. we allow overriding `shouldForwardProp` completely by spreading `nextOptions` in here - so this isn't composing it

I definitely feel like 1. should get fixed. Still unsure what the expectation for the 2. is though. When using `as` we keep the configured `shouldForwardProp` (if it got configured), but use what would be the default for a new component (the one in `as`) if `shouldForwardProp` was not configured. This API is quirky 🤷‍♂ 

Thoughts?

Reply:
I am going to mark this comment as resolved as the base branch changed and messed up this audit -- In my recent push i've addressed both points that hopefully satisfy things.